src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/bfaction/BfActionReportTest.java

```diff
@@ -0,0 +1,282 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.reports.bfaction;
+
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.ACCEPTED_STATE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.BROUGHT_FORWARD_REPORT;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.LEEDS_LISTING_CASE_TYPE_ID;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.RANGE_HEARING_DATE_TYPE;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SINGLE_CASE_TYPE;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.platform.commons.util.StringUtils;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.items.BFActionTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.types.BFActionType;
+import uk.gov.hmcts.ecm.common.model.ccd.types.CasePreAcceptType;
+import uk.gov.hmcts.ecm.common.model.listing.ListingData;
+import uk.gov.hmcts.ecm.common.model.listing.ListingDetails;
+
+public class BfActionReportTest {
+    private List<SubmitEvent> submitEvents;
+    private ListingDetails listingDetails;
+    private ListingData listingData;
+
+    @Before
+    public void setUp() {
+        listingDetails = new ListingDetails();
+        listingData = new ListingData();
+        listingData.setReportType(BROUGHT_FORWARD_REPORT);
+        listingDetails.setJurisdiction("EMPLOYMENT");
+        listingData.setListingVenue("Leeds");
+        listingDetails.setCaseTypeId(LEEDS_LISTING_CASE_TYPE_ID);
+        submitEvents = new ArrayList<>();
+        listingData = new ListingData();
+        listingData.setListingDateFrom("2019-12-08");
+        listingData.setListingDateTo("2019-12-20");
+        listingData.setHearingDateType(RANGE_HEARING_DATE_TYPE);
+        listingDetails.setCaseData(listingData);
+    }
+
+    @Test
+    public void shouldReturnOnlyOpenBfActionsWithInDateRange() {
+        var caseData = new CaseData();
+        caseData.setEthosCaseReference("1800522/2020");
+        caseData.setReceiptDate("2018-08-10");
+        var casePreAcceptType2 = new CasePreAcceptType();
+        casePreAcceptType2.setDateAccepted("2018-08-10");
+        caseData.setPreAcceptCase(casePreAcceptType2);
+        caseData.setEcmCaseType(SINGLE_CASE_TYPE);
```

Comment:
Isn't all of this caseData setting the same as your getTestData method?

Reply:
refactored it.